flutter-for-ios.md

```diff
@@ -27,3 +27,429 @@ that are most relevant to your needs.
 
 * TOC Placeholder
 {:toc}
+
+# Views 
+
+## What is the equivalent of a `UIView` in Flutter?
+
+On iOS, the majority of anything you may want to create in the UI is done using
+view objects, which are instances of the `UIView` class.  These can act as
+containers for other `UIView` classes, which form your layout.
+
+In Flutter, the equivalent to this is a `Widget`. However, these have a few
+differences to a  `UIView`. To start, widgets only exist for one frame, and on
+every new frame, Flutter’s framework creates a new tree of widget instances.
+In comparison, an iOS view is drawn once and does not redraw until it is
+invalidated using `setNeedsDisplay()`.
+
+Furthermore, unlike `UIView`, Flutter’s widgets are immutable, which allows
+them to be very lightweight. By default, Flutter’s widgets are styled as
+‘Material’ (i.e. Android themed), however you can use [Cupertino
+widgets](https://flutter.io/widgets/cupertino/) to specify an iOS theme if you
+require.
+
+## How do I update `Widget`s?
+
+To update your views on iOS you directly mutate them. In Flutter, widgets are
+immutable and not updated directly. Instead, you have to manipulate the
+widget’s state. This is where the concept of Stateful vs Stateless widgets
+comes in. A `StatelessWidget` is just what it sounds like - a widget with no
+state attached.
+
+`StatelessWidgets` are useful when the part of the user interface you are
+describing does not depend on anything other than the initial configuration
+information in the widget.
+
+For example, in iOS, this would be similar to placing a `UIImageView` with
+your logo as the `image`. If the logo is not going to change during runtime,
+you would use a `StatelessWidget` in Flutter.
+
+If you want to dynamically change the UI based on data received after making an
+HTTP call then you would use a `StatefulWidget`. After the HTTP call had
+complete, you would tell the Flutter framework that the widget’s `State` has
+been updated so it could update the widget.
+
+The important thing to note here is at the core both stateless and stateful
+widgets behave the same i.e. they rebuild every frame. The difference is that
```

Comment:
Only if necessary. Flutter doesn't run the build() method on every frame. Only when it needs to redraw the widget.